commit 9fbce8e744c23975e8d310e403c35094a84a531f
Author: dann frazier <dannf@hp.com>
Date:   Mon Feb 15 21:40:34 2010 -0700

    [ Adjusted to apply to Debian's 2.6.18 ]
    commit 6f5a55f1a6c5abee15a0e878e5c74d9f1569b8b0
    Author: Linus Torvalds <torvalds@linux-foundation.org>
    Date:   Fri Feb 5 16:16:50 2010 -0800
    
       Fix potential crash with sys_move_pages

diff --git a/mm/migrate.c b/mm/migrate.c
index 01d0274..bf8559f 100644
--- a/mm/migrate.c
+++ b/mm/migrate.c
@@ -943,6 +943,9 @@ SYSCALL_DEFINE6(move_pages, pid_t, pid, unsigned long, nr_pages,
 				goto out;
 
 			err = -ENODEV;
+			if (node < 0 || node >= MAX_NUMNODES)
+				goto out;
+
 			if (!node_online(node))
 				goto out;
 
